fixups/RegLegacyFixups/InitializeFixup.cpp

```diff
@@ -0,0 +1,238 @@
+        std::size_t count = std::vsnprintf(str.data(), str.size() + 1, fmt, args);
+        assert(count >= 0);
+        va_end(args);
+
+        if (count > str.size())
+        {
+            count = 1024;       // vswprintf actually returns a negative number, let's just go with something big enough for our long strings; it is resized shortly.
+            str.resize(count);
+
+            va_list args2;
+            va_start(args2, fmt);
+            count = std::vsnprintf(str.data(), str.size() + 1, fmt, args2);
+            assert(count >= 0);
+            va_end(args2);
+        }
+
+        str.resize(count);
+#if _DEBUG
+        ::OutputDebugStringA(str.c_str());
+#endif
+    }
+    catch (...)
+    {
+        ::OutputDebugStringA("Exception in Log()");
+        ::OutputDebugStringA(fmt);
+    }
+}
+void Log(const wchar_t* fmt, ...)
+{
+    try
+    {
+        va_list args;
+        va_start(args, fmt);
+
+        std::wstring wstr;
+        wstr.resize(256);
+        std::size_t count = std::vswprintf(wstr.data(), wstr.size() + 1, fmt, args);
+        va_end(args);
+
+        if (count > wstr.size())
+        {
+            count = 1024;       // vswprintf actually returns a negative number, let's just go with something big enough for our long strings; it is resized shortly.
+            wstr.resize(count);
+            va_list args2;
+            va_start(args2, fmt);
+            count = std::vswprintf(wstr.data(), wstr.size() + 1, fmt, args2);
+            va_end(args2);
+        }
+        wstr.resize(count);
+#if _DEBUG
+        ::OutputDebugStringW(wstr.c_str());
+#endif
+    }
+    catch (...)
+    {
+        ::OutputDebugStringA("Exception in wide Log()");
+        ::OutputDebugStringW(fmt);
+    }
+}
+void LogString(const char* name, const char* value)
+{
+    Log("%s=%s\n", name, value);
+}
+void LogString(const char* name, const wchar_t* value)
+{
+    Log("%s=%ls\n", name, value);
+}
+void LogString(const wchar_t* name, const char* value)
+{
+    Log("%ls=%s\n", name, value);
+}
+void LogString(const wchar_t* name, const wchar_t* value)
+{
+    Log(L"%ls=%ls\n", name, value);
+}
+
+
+void InitializeFixups()
+{
+
+    Log("Initializing RegLegacyFixups\n");
+    
+}
+
+
+void InitializeConfiguration()
+{
+    Log("RegLegacyFixups Start InitializeConfiguration()\n");
+    
+    if (auto rootConfig = ::PSFQueryCurrentDllConfig())
+    {
+        if (rootConfig != NULL)
+        {
+            Log("RegLegacyFixups process config\n");
+            const psf::json_array& rootConfigArray = rootConfig->as_array();
+            for (auto& spec : rootConfigArray)
+            {
+                Log("RegLegacyFixups: process spec\n");
+
+                Reg_Remediation_Spec specItem;
+
+                auto& specObject = spec.as_object();
+                Log("RegLegacyFixups: Have specObject\n");
+
+                auto type = specObject.get("type").as_string().wstring();
+                Log("RegLegacyFixups: have type");
+#if _DEBUG
```

Comment:
Can you please move all the #IF _DEBUG calls to one method and call the logging function regardless if we are in debug or release.

Reply:
done.